---
title: "SLE712 Assessment 3"
author: "Nishat Nini Urmi"
date: "6/7/2020"
output: html_document
editor_options: 
  chunk_output_type: inline
---
 
### Part 1: Importing files, data wrangling, mathematical operations, plots and saving code on GitHub 

##Question 1 : Read in the file, making the gene accession numbers the row names. Show a table of values for the first six genes
###Answer: 
```{r}
download.file("https://raw.githubusercontent.com/markziemann/SLE712_files/master/bioinfo_asst3_part1_files/gene_expression.tsv", destfile = "gene_expression.tsv" )
x<- read.table("data/gene_expression.tsv", header = TRUE, stringsAsFactors = FALSE, row.names = 1)
head(x)
str(x)
x[1:6,1:2]
```
The row names were changed into gene accession number while naming the table as x
##Question 2 : Make a new column which is the mean of the other columns. Show a table of values for the first six genes
###Answer: 
```{r}
x$Mean_cal <- rowMeans(x)
head(x)
x[1:6,1:3]
```
A new column "Mean_cal" is created that holds the mean value of the other columns. 

##Question3. List the 10 genes with the highest mean expression 
###Answer: 
```{r}
Highest_mean <-x[order(-x$Mean_cal),]
head(Highest_mean,10)
```

##Question 4.Determine the number of genes with a mean <10 
###Answer: 
```{r}
filteredmean <- subset(x,Mean_cal<10)
nrow(filteredmean)
```

##Question 5. Make a histogram plot of the mean values in png format and paste it into your report
###Answer:
```{r,"Histogram for Mean values",echo=FALSE}
hist(x$Mean_cal,main="Histogram for Mean values",xlab = "mean value",border = "red",col="blue",breaks = 50,xlim=c(0,10000))
```
##Question 6. Import this csv file into an R object. What are the column names?
###Answer:The CSV files were uploaded from the link  The "csv" file was saved as obejct "y". 
```{r}
download.file("https://raw.githubusercontent.com/markziemann/SLE712_files/master/bioinfo_asst3_part1_files/growth_data.csv",destfile = "growth_data.csv")
y <- read.csv("data/growth_data.csv", header = TRUE, stringsAsFactors = FALSE)
head(y)
str(y)
```
###Name of the columns are- "site", "TreeID","Circumf_2004_cm", "Circumf_2009_cm", "Circumf_2014_cm " and "Circumf_2019_cm"

##Question 7 Calculate the mean and standard deviation of tree circumference at the start and end of the study at both sites. 

###Answer: Prior to calculating mean and Standard deviation, both sites of study were saved as seperate obejcts "NE" for "NorthEast" and "SE" for "SouthWest" 
```{r}

NE <- subset(y,Site=="northeast")
SW <- subset(y,Site=="southwest")
head(NE)
head(SW)
```

###Mean calculation of tree circumference at the start and end of the study for North East Site
```{r}
mean(NE$Circumf_2004_cm)
mean(NE$Circumf_2019_cm)
sd(NE$Circumf_2004_cm)
sd(NE$Circumf_2019_cm)
```
Mean  and standard deviation calculation of tree circumfurence at the satrt and end of the site Southwest
```{r}
mean(SW$Circumf_2004_cm)
mean(SW$Circumf_2019_cm)
sd(SW$Circumf_2004_cm)
sd(SW$Circumf_2019_cm)
```
##Question 8 Make a box plot of tree circumference at the start and end of the study at both sites.
```{r}
boxplot(NE$Circumf_2004_cm,NE$Circumf_2019_cm,SW$Circumf_2004_cm,SW$Circumf_2019_cm,names=c("NE2004","NE2019","SW2004","SW2019"),ylab="Circumference(cm)",main="Growth at 2 plantation sites")
```

##Question 9  Calculate the mean growth over the past 10 years at each site.
###Answer: For calculating mean growth for past 10 yeasrs in each site, two seperate columns were created.
```{r}
NE$Growth <- (NE$Circumf_2019_cm-NE$Circumf_2009_cm )
head(NE)
```
```{r}

SW$Growth <- (SW$Circumf_2019_cm-SW$Circumf_2009_cm)
head(SW)
```

##Question 10 Use the t.test and wilcox.test functions to estimate the p-value that the 10 year growth is different at the two site
###Answer: For t.test and wilcox.test, the columns that demonstrated meam growth values for past 10 years

**t.test**
```{r}
t.test(SW$Growth,NE$Growth)
```
**wolcox.test**
```{r}
wilcox.test(SW$Growth,NE$Growth)
```
##the P-value for t-test is 1.713e-06, and for Wilcox test it is 4.626e-06

-----

### Part 2 : Determine the limits of BLAST 

Required Libraries and sources 
Some packages need to be downloaded to work with sequence data. The packages are-
 `seqinr` to process and analyse sequence data
```{r}
library("seqinr")
```
 `R.utils`to extract compressed files
```{r}
library("R.utils")
```
 `rBLAST`for running BLAST searches
```{r}
library("rBLAST")
```
 `ape`cluster alighment property
```{r}
library("ape")
```
 `ORFik`analysis of open reading frames in the genome of interest or set of transcripts 
```{r}
library("ORFik")
```
 `Biostrings`efficient manipulation of biological strings
```{r}
library("Biostrings")
```
 
##Question 1  Download the whole set of E. coli gene DNA sequences and use gunzip to decompress. Use the makeblast() function to create a blast database. How many sequences are present in the _E.coli_ set? 
##Answer:
```{r}
download.file("ftp://ftp.ensemblgenomes.org/pub/bacteria/release-42/fasta/bacteria_0_collection/escherichia_coli_str_k_12_substr_mg1655/cds/Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.cds.all.fa.gz",
              destfile = "Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.cds.all.fa.gz")

### then dicompressed the sequence and saved it by using following function 
R.utils::gunzip("Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.cds.all.fa.gz", overwrite= TRUE)

makeblastdb("Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.cds.all.fa",dbtype = "nucl","-parse_seqids")
```
The _E.coli_ set holds 4140 number of sequences 

##Question 2 Download the sample fasta sequences and read them in as above. For your allocated sequence, determine the length (in bp) and the proportion of GC bases
###Answer:
```{r}
if ( ! file.exists("sample.fa") ) {
  download.file("https://raw.githubusercontent.com/markziemann/SLE712_files/master/bioinfo_asst3_part2_files/sample.fa",
                destfile = "sample.fa")
 
}
Ecoli_Sample <- read.fasta("sample.fa")
```
determines the allocated sequence and saved it as an object 
```{r}
allocated_Seq <- Ecoli_Sample$`53`
```
For calculating the length in bp and determinig number of each base pair  
```{r}
seqinr::getLength(allocated_Seq)
table(allocated_Seq)
```
For GC contents
```{r}
seqinr::GC(allocated_Seq)
```
The length of the sequence is 789 and the GC proportion is 54.24588% 

###Question 3 You will be provided with R functions to create BLAST databases and perform blast searches. Use blast to identify what _E. coli_ gene your sequence matches best. Show a table of the top 3 hits including percent identity, E-value and bit scores. 4 
##Answer:
```{r}
source("https://raw.githubusercontent.com/markziemann/SLE712_files/master/bioinfo_asst3_part2_files/mutblast_functions.R")
```
Blast search 
```{r}
res <- myblastn_tab(allocated_Seq, db="Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.cds.all.fa")
res
```
We can see that the allocated sequence parfectly matches the Source _E.coli_ sequence. The sequence is "AAC76075". 

#The function to calculate the top 3 hit 
```{r}
hits <- as.character(res$sseqid[1:3])
hits
```
The percent Identity is 100, E- value 0 and Bitscore - 1517

##Question 4 You will be provided with a function that enables you to make a set number of point mutations to your sequence of interest. Run the function and write an R code to check the number of mismatches between the original and mutated sequence. 

##Answer: create a copy of my allocated sequnce that has certain number of mismathces in it
```{r}
seqinr::write.fasta(allocated_Seq,names="allocated_Seq",file.out = "allocated_Seq.fa")
makeblastdb("allocated_Seq.fa",dbtype="nucl", "-parse_seqids")
res <- myblastn_tab(allocated_Seq, db="allocated_Seq.fa")
res
```
changing the number to check the mismatches 
```{r}
my_allocated_mutator<-mutator(allocated_Seq,20)
res <- myblastn_tab(my_allocated_mutator, db="allocated_Seq.fa")
res
```

##Question 5 Using the provided functions for mutating and BLASTing a sequence, determine the number and proportion of sites that need to be altered to prevent the BLAST search from matching the gene of origin. Because the mutation is random, you may need to run this test multiple times to get a reliable answer

##Answer : At 1st we need to create a function that will do mutation and blast and give the result as 0 or 1 depending on the blast was successful or not
```{r}
myfunc <- function(myseq,nmut) { 
  mutseq <- mutator( myseq= allocated_Seq, nmut = nmut) #the sequence for mutation, nmut=nmut as it will recognize the number after the sequence as nmut 
  res <- myblastn_tab(myseq= allocated_Seq,db= "Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.cds.all.fa") #for blast
  if (is.null(res)) {myres= 0} else {myres = 1} 
  return(myres) 
}
```
 applying the created function 
```{r}
myfunc(myseq,100)
```
Then applying the replicate fucntion that will run the same function several times and that would give a vector of number of values 
```{r}
replicate(n = 100, myfunc(myseq,100))
```
geting the summerised result 
```{r}
mean(replicate(100,myfunc(myseq,100 )))
```
we need to run this for all the values for nmut. The numbers are saved as n below which is used to run the replicate command 
```{r}
n <-c (0,100,200,300,400,500,600,700)
myfunction_rep <- function(nmut) {
 mean(replicate(n=100, myfunc(myseq,100)))
}
```
The final answer with sapply comand 
```{r}
finalress <- sapply( n ,myfunction_rep)
finalress
```

