\PassOptionsToPackage{unicode=true}{hyperref} % options for packages loaded elsewhere
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provides euro and other symbols
\else % if luatex or xelatex
  \usepackage{unicode-math}
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage[]{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\usepackage{hyperref}
\hypersetup{
            pdfauthor={Nishat Nini Urmi},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

% set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother


\author{Nishat Nini Urmi}
\date{12/6/2020}

\begin{document}

\hypertarget{part-1-importing-files-data-wrangling-mathematical-operations-plots-and-saving-code-on-github}{%
\subsubsection{Part 1: Importing files, data wrangling, mathematical
operations, plots and saving code on
GitHub}\label{part-1-importing-files-data-wrangling-mathematical-operations-plots-and-saving-code-on-github}}

\#\#Question 1 : Read in the file, making the gene accession numbers the
row names. Show a table of values for the first six genes \#\#\#Answer:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{download.file}\NormalTok{(}\StringTok{"https://raw.githubusercontent.com/markziemann/SLE712_files/master/bioinfo_asst3_part1_files/gene_expression.tsv"}\NormalTok{, }\DataTypeTok{destfile =} \StringTok{"gene_expression.tsv"}\NormalTok{ )}
\NormalTok{x<-}\StringTok{ }\KeywordTok{read.table}\NormalTok{(}\StringTok{"data/gene_expression.tsv"}\NormalTok{, }\DataTypeTok{header =} \OtherTok{TRUE}\NormalTok{, }\DataTypeTok{stringsAsFactors =} \OtherTok{FALSE}\NormalTok{, }\DataTypeTok{row.names =} \DecValTok{1}\NormalTok{)}
\KeywordTok{head}\NormalTok{(x)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                 SRR5150592 SRR5150593
## ENSG00000223972          1          0
## ENSG00000227232          0          1
## ENSG00000278267          0          0
## ENSG00000243485          0          0
## ENSG00000284332          0          0
## ENSG00000237613          0          0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{str}\NormalTok{(x)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 'data.frame':    58302 obs. of  2 variables:
##  $ SRR5150592: int  1 0 0 0 0 0 0 0 0 0 ...
##  $ SRR5150593: int  0 1 0 0 0 0 0 0 0 0 ...
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{x[}\DecValTok{1}\OperatorTok{:}\DecValTok{6}\NormalTok{,}\DecValTok{1}\OperatorTok{:}\DecValTok{2}\NormalTok{]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                 SRR5150592 SRR5150593
## ENSG00000223972          1          0
## ENSG00000227232          0          1
## ENSG00000278267          0          0
## ENSG00000243485          0          0
## ENSG00000284332          0          0
## ENSG00000237613          0          0
\end{verbatim}

The row names were changed into gene accession number while naming the
table as x \#\#Question 2 : Make a new column which is the mean of the
other columns. Show a table of values for the first six genes
\#\#\#Answer:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{x}\OperatorTok{$}\NormalTok{Mean_cal <-}\StringTok{ }\KeywordTok{rowMeans}\NormalTok{(x)}
\KeywordTok{head}\NormalTok{(x)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                 SRR5150592 SRR5150593 Mean_cal
## ENSG00000223972          1          0      0.5
## ENSG00000227232          0          1      0.5
## ENSG00000278267          0          0      0.0
## ENSG00000243485          0          0      0.0
## ENSG00000284332          0          0      0.0
## ENSG00000237613          0          0      0.0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{x[}\DecValTok{1}\OperatorTok{:}\DecValTok{6}\NormalTok{,}\DecValTok{1}\OperatorTok{:}\DecValTok{3}\NormalTok{]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                 SRR5150592 SRR5150593 Mean_cal
## ENSG00000223972          1          0      0.5
## ENSG00000227232          0          1      0.5
## ENSG00000278267          0          0      0.0
## ENSG00000243485          0          0      0.0
## ENSG00000284332          0          0      0.0
## ENSG00000237613          0          0      0.0
\end{verbatim}

A new column ``Mean\_cal'' is created that holds the mean value of the
other columns.

\#\#Question3. List the 10 genes with the highest mean expression
\#\#\#Answer:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Highest_mean <-x[}\KeywordTok{order}\NormalTok{(}\OperatorTok{-}\NormalTok{x}\OperatorTok{$}\NormalTok{Mean_cal),]}
\KeywordTok{head}\NormalTok{(Highest_mean,}\DecValTok{10}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                 SRR5150592 SRR5150593 Mean_cal
## ENSG00000115414     311857     206347 259102.0
## ENSG00000210082     145916     163288 154602.0
## ENSG00000075624     133983     116762 125372.5
## ENSG00000198886      91596      99943  95769.5
## ENSG00000137801      95158      74546  84852.0
## ENSG00000198804      79832      84774  82303.0
## ENSG00000198786      74570      83589  79079.5
## ENSG00000196924      88225      66413  77319.0
## ENSG00000198712      76108      77108  76608.0
## ENSG00000108821      80342      60127  70234.5
\end{verbatim}

\#\#Question 4.Determine the number of genes with a mean \textless{}10
\#\#\#Answer:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{filteredmean <-}\StringTok{ }\KeywordTok{subset}\NormalTok{(x,Mean_cal}\OperatorTok{<}\DecValTok{10}\NormalTok{)}
\KeywordTok{nrow}\NormalTok{(filteredmean)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 43124
\end{verbatim}

\#\#Question 5. Make a histogram plot of the mean values in png format
and paste it into your report \#\#\#Answer:
\includegraphics{rmd_files/figure-latex/Histogram for Mean values-1.pdf}
\#\#Question 6. Import this csv file into an R object. What are the
column names? \#\#\#Answer:The CSV files were uploaded from the link The
``csv'' file was saved as obejct ``y''.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{download.file}\NormalTok{(}\StringTok{"https://raw.githubusercontent.com/markziemann/SLE712_files/master/bioinfo_asst3_part1_files/growth_data.csv"}\NormalTok{,}\DataTypeTok{destfile =} \StringTok{"growth_data.csv"}\NormalTok{)}
\NormalTok{y <-}\StringTok{ }\KeywordTok{read.csv}\NormalTok{(}\StringTok{"data/growth_data.csv"}\NormalTok{, }\DataTypeTok{header =} \OtherTok{TRUE}\NormalTok{, }\DataTypeTok{stringsAsFactors =} \OtherTok{FALSE}\NormalTok{)}
\KeywordTok{head}\NormalTok{(y)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##        Site TreeID Circumf_2004_cm Circumf_2009_cm Circumf_2014_cm
## 1 northeast   A003             5.2            10.1            19.9
## 2 northeast   A005             4.9             9.6            18.9
## 3 northeast   A007             3.7             7.3            14.3
## 4 northeast   A008             3.8             6.5            10.9
## 5 northeast   A011             3.8             6.4            10.9
## 6 northeast   A012             5.9            10.0            16.8
##   Circumf_2019_cm
## 1            38.9
## 2            37.0
## 3            28.1
## 4            18.5
## 5            18.4
## 6            28.4
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{str}\NormalTok{(y)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 'data.frame':    100 obs. of  6 variables:
##  $ Site           : chr  "northeast" "northeast" "northeast" "northeast" ...
##  $ TreeID         : chr  "A003" "A005" "A007" "A008" ...
##  $ Circumf_2004_cm: num  5.2 4.9 3.7 3.8 3.8 5.9 4.4 5.3 7.1 3.8 ...
##  $ Circumf_2009_cm: num  10.1 9.6 7.3 6.5 6.4 10 9.9 9 12 7.4 ...
##  $ Circumf_2014_cm: num  19.9 18.9 14.3 10.9 10.9 16.8 22.2 15.2 20.2 14.5 ...
##  $ Circumf_2019_cm: num  38.9 37 28.1 18.5 18.4 28.4 50 25.8 34.2 28.4 ...
\end{verbatim}

\#\#\#Name of the columns are- ``site'',
``TreeID'',``Circumf\_2004\_cm'', ``Circumf\_2009\_cm'',
``Circumf\_2014\_cm'' and ``Circumf\_2019\_cm''

\#\#Question 7 Calculate the mean and standard deviation of tree
circumference at the start and end of the study at both sites.

\#\#\#Answer: Prior to calculating mean and Standard deviation, both
sites of study were saved as seperate obejcts ``NE'' for ``NorthEast''
and ``SE'' for ``SouthWest''

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{NE <-}\StringTok{ }\KeywordTok{subset}\NormalTok{(y,Site}\OperatorTok{==}\StringTok{"northeast"}\NormalTok{)}
\NormalTok{SW <-}\StringTok{ }\KeywordTok{subset}\NormalTok{(y,Site}\OperatorTok{==}\StringTok{"southwest"}\NormalTok{)}
\KeywordTok{head}\NormalTok{(NE)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##        Site TreeID Circumf_2004_cm Circumf_2009_cm Circumf_2014_cm
## 1 northeast   A003             5.2            10.1            19.9
## 2 northeast   A005             4.9             9.6            18.9
## 3 northeast   A007             3.7             7.3            14.3
## 4 northeast   A008             3.8             6.5            10.9
## 5 northeast   A011             3.8             6.4            10.9
## 6 northeast   A012             5.9            10.0            16.8
##   Circumf_2019_cm
## 1            38.9
## 2            37.0
## 3            28.1
## 4            18.5
## 5            18.4
## 6            28.4
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{head}\NormalTok{(SW)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##         Site TreeID Circumf_2004_cm Circumf_2009_cm Circumf_2014_cm
## 51 southwest   A001             5.3            13.5            34.6
## 52 southwest   A002             5.2            10.1            19.8
## 53 southwest   A004             6.2            15.9            40.6
## 54 southwest   A006             5.1            11.5            25.9
## 55 southwest   A009             3.6             9.1            23.4
## 56 southwest   A010             6.6            14.9            33.6
##    Circumf_2019_cm
## 51            88.7
## 52            38.8
## 53           103.9
## 54            58.3
## 55            59.8
## 56            75.5
\end{verbatim}

\#\#\#Mean calculation of tree circumference at the start and end of the
study for North East Site

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(NE}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2004}\NormalTok{_cm)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 5.078
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(NE}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2019}\NormalTok{_cm)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 40.052
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(NE}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2004}\NormalTok{_cm)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 1.059127
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(NE}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2019}\NormalTok{_cm)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 16.90443
\end{verbatim}

Mean and standard deviation calculation of tree circumfurence at the
satrt and end of the site Southwest

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(SW}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2004}\NormalTok{_cm)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 5.076
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(SW}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2019}\NormalTok{_cm)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 59.772
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(SW}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2004}\NormalTok{_cm)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 1.060527
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sd}\NormalTok{(SW}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2019}\NormalTok{_cm)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 22.57784
\end{verbatim}

\#\#Question 8 Make a box plot of tree circumference at the start and
end of the study at both sites.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{boxplot}\NormalTok{(NE}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2004}\NormalTok{_cm,NE}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2019}\NormalTok{_cm,SW}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2004}\NormalTok{_cm,SW}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2019}\NormalTok{_cm,}\DataTypeTok{names=}\KeywordTok{c}\NormalTok{(}\StringTok{"NE2004"}\NormalTok{,}\StringTok{"NE2019"}\NormalTok{,}\StringTok{"SW2004"}\NormalTok{,}\StringTok{"SW2019"}\NormalTok{),}\DataTypeTok{ylab=}\StringTok{"Circumference(cm)"}\NormalTok{,}\DataTypeTok{main=}\StringTok{"Growth at 2 plantation sites"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{rmd_files/figure-latex/unnamed-chunk-9-1.pdf}

\#\#Question 9 Calculate the mean growth over the past 10 years at each
site. \#\#\#Answer: For calculating mean growth for past 10 years in
each site, two seperate columns were created.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{NE}\OperatorTok{$}\NormalTok{Growth <-}\StringTok{ }\NormalTok{(NE}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2019}\NormalTok{_cm}\OperatorTok{-}\NormalTok{NE}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2009}\NormalTok{_cm )}
\KeywordTok{head}\NormalTok{(NE)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##        Site TreeID Circumf_2004_cm Circumf_2009_cm Circumf_2014_cm
## 1 northeast   A003             5.2            10.1            19.9
## 2 northeast   A005             4.9             9.6            18.9
## 3 northeast   A007             3.7             7.3            14.3
## 4 northeast   A008             3.8             6.5            10.9
## 5 northeast   A011             3.8             6.4            10.9
## 6 northeast   A012             5.9            10.0            16.8
##   Circumf_2019_cm Growth
## 1            38.9   28.8
## 2            37.0   27.4
## 3            28.1   20.8
## 4            18.5   12.0
## 5            18.4   12.0
## 6            28.4   18.4
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{SW}\OperatorTok{$}\NormalTok{Growth <-}\StringTok{ }\NormalTok{(SW}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2019}\NormalTok{_cm}\OperatorTok{-}\NormalTok{SW}\OperatorTok{$}\NormalTok{Circumf_}\DecValTok{2009}\NormalTok{_cm)}
\KeywordTok{head}\NormalTok{(SW)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##         Site TreeID Circumf_2004_cm Circumf_2009_cm Circumf_2014_cm
## 51 southwest   A001             5.3            13.5            34.6
## 52 southwest   A002             5.2            10.1            19.8
## 53 southwest   A004             6.2            15.9            40.6
## 54 southwest   A006             5.1            11.5            25.9
## 55 southwest   A009             3.6             9.1            23.4
## 56 southwest   A010             6.6            14.9            33.6
##    Circumf_2019_cm Growth
## 51            88.7   75.2
## 52            38.8   28.7
## 53           103.9   88.0
## 54            58.3   46.8
## 55            59.8   50.7
## 56            75.5   60.6
\end{verbatim}

\#\#Question 10 Use the t.test and wilcox.test functions to estimate the
p-value that the 10 year growth is different at the two site
\#\#\#Answer: For t.test and wilcox.test, the columns that demonstrated
mean growth values for past 10 years

\textbf{t.test}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{t.test}\NormalTok{(SW}\OperatorTok{$}\NormalTok{Growth,NE}\OperatorTok{$}\NormalTok{Growth)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Welch Two Sample t-test
## 
## data:  SW$Growth and NE$Growth
## t = 5.124, df = 89.366, p-value = 1.713e-06
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  11.19057 25.36543
## sample estimates:
## mean of x mean of y 
##    48.354    30.076
\end{verbatim}

\textbf{wolcox.test}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{wilcox.test}\NormalTok{(SW}\OperatorTok{$}\NormalTok{Growth,NE}\OperatorTok{$}\NormalTok{Growth)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  SW$Growth and NE$Growth
## W = 1915, p-value = 4.626e-06
## alternative hypothesis: true location shift is not equal to 0
\end{verbatim}

\#\#the P-value for t-test is 1.713e-06, and for Wilcox test it is
4.626e-06

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

\hypertarget{part-2-determine-the-limits-of-blast}{%
\subsubsection{Part 2 : Determine the limits of
BLAST}\label{part-2-determine-the-limits-of-blast}}

Required Libraries and sources Some packages need to be downloaded to
work with sequence data. The packages are- \texttt{seqinr} to process
and analyse sequence data

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(}\StringTok{"seqinr"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\texttt{R.utils}to extract compressed files

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(}\StringTok{"R.utils"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: R.oo
\end{verbatim}

\begin{verbatim}
## Loading required package: R.methodsS3
\end{verbatim}

\begin{verbatim}
## R.methodsS3 v1.8.0 (2020-02-14 07:10:20 UTC) successfully loaded. See ?R.methodsS3 for help.
\end{verbatim}

\begin{verbatim}
## R.oo v1.23.0 successfully loaded. See ?R.oo for help.
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'R.oo'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:R.methodsS3':
## 
##     throw
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:seqinr':
## 
##     getName
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:methods':
## 
##     getClasses, getMethods
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:base':
## 
##     attach, detach, load, save
\end{verbatim}

\begin{verbatim}
## R.utils v2.9.2 successfully loaded. See ?R.utils for help.
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'R.utils'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:utils':
## 
##     timestamp
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:base':
## 
##     cat, commandArgs, getOption, inherits, isOpen, nullfile, parse,
##     warnings
\end{verbatim}

\texttt{rBLAST}for running BLAST searches

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(}\StringTok{"rBLAST"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: Biostrings
\end{verbatim}

\begin{verbatim}
## Loading required package: BiocGenerics
\end{verbatim}

\begin{verbatim}
## Loading required package: parallel
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'BiocGenerics'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which, which.max, which.min
\end{verbatim}

\begin{verbatim}
## Loading required package: S4Vectors
\end{verbatim}

\begin{verbatim}
## Loading required package: stats4
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'S4Vectors'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:base':
## 
##     expand.grid
\end{verbatim}

\begin{verbatim}
## Loading required package: IRanges
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'IRanges'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:R.oo':
## 
##     trim
\end{verbatim}

\begin{verbatim}
## Loading required package: XVector
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'Biostrings'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:seqinr':
## 
##     translate
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:base':
## 
##     strsplit
\end{verbatim}

\texttt{ape}cluster alighment property

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(}\StringTok{"ape"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Attaching package: 'ape'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:Biostrings':
## 
##     complement
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:seqinr':
## 
##     as.alignment, consensus
\end{verbatim}

\texttt{ORFik}analysis of open reading frames in the genome of interest
or set of transcripts

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(}\StringTok{"ORFik"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: GenomicRanges
\end{verbatim}

\begin{verbatim}
## Loading required package: GenomeInfoDb
\end{verbatim}

\begin{verbatim}
## Loading required package: GenomicAlignments
\end{verbatim}

\begin{verbatim}
## Loading required package: SummarizedExperiment
\end{verbatim}

\begin{verbatim}
## Loading required package: Biobase
\end{verbatim}

\begin{verbatim}
## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.
\end{verbatim}

\begin{verbatim}
## Loading required package: DelayedArray
\end{verbatim}

\begin{verbatim}
## Loading required package: matrixStats
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'matrixStats'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:Biobase':
## 
##     anyMissing, rowMedians
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:seqinr':
## 
##     count
\end{verbatim}

\begin{verbatim}
## Loading required package: BiocParallel
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'DelayedArray'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:matrixStats':
## 
##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:base':
## 
##     aperm, apply, rowsum
\end{verbatim}

\begin{verbatim}
## Loading required package: Rsamtools
\end{verbatim}

\begin{verbatim}
## Registered S3 method overwritten by 'GGally':
##   method from   
##   +.gg   ggplot2
\end{verbatim}

\texttt{Biostrings}efficient manipulation of biological strings

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(}\StringTok{"Biostrings"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\#\#Question 1 Download the whole set of E. coli gene DNA sequences and
use gunzip to decompress. Use the makeblast() function to create a blast
database. How many sequences are present in the \emph{E.coli} set?
\#\#Answer:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{download.file}\NormalTok{(}\StringTok{"ftp://ftp.ensemblgenomes.org/pub/bacteria/release-42/fasta/bacteria_0_collection/escherichia_coli_str_k_12_substr_mg1655/cds/Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.cds.all.fa.gz"}\NormalTok{,}
              \DataTypeTok{destfile =} \StringTok{"Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.cds.all.fa.gz"}\NormalTok{)}

\CommentTok{### then dicompressed the sequence and saved it by using following function }
\NormalTok{R.utils}\OperatorTok{::}\KeywordTok{gunzip}\NormalTok{(}\StringTok{"Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.cds.all.fa.gz"}\NormalTok{, }\DataTypeTok{overwrite=} \OtherTok{TRUE}\NormalTok{)}

\KeywordTok{makeblastdb}\NormalTok{(}\StringTok{"Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.cds.all.fa"}\NormalTok{,}\DataTypeTok{dbtype =} \StringTok{"nucl"}\NormalTok{,}\StringTok{"-parse_seqids"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

The \emph{E.coli} set holds 4140 number of sequences

\#\#Question 2 Download the sample fasta sequences and read them in as
above. For your allocated sequence, determine the length (in bp) and the
proportion of GC bases \#\#\#Answer:

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{if}\NormalTok{ ( }\OperatorTok{!}\StringTok{ }\KeywordTok{file.exists}\NormalTok{(}\StringTok{"sample.fa"}\NormalTok{) ) \{}
  \KeywordTok{download.file}\NormalTok{(}\StringTok{"https://raw.githubusercontent.com/markziemann/SLE712_files/master/bioinfo_asst3_part2_files/sample.fa"}\NormalTok{,}
                \DataTypeTok{destfile =} \StringTok{"sample.fa"}\NormalTok{)}
 
\NormalTok{\}}
\NormalTok{Ecoli_Sample <-}\StringTok{ }\KeywordTok{read.fasta}\NormalTok{(}\StringTok{"sample.fa"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

determines the allocated sequence and saved it as an object

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{allocated_Seq <-}\StringTok{ }\NormalTok{Ecoli_Sample}\OperatorTok{$}\StringTok{`}\DataTypeTok{53}\StringTok{`}
\end{Highlighting}
\end{Shaded}

For calculating the length in bp and determinig number of each base pair

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{seqinr}\OperatorTok{::}\KeywordTok{getLength}\NormalTok{(allocated_Seq)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 789
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{table}\NormalTok{(allocated_Seq)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## allocated_Seq
##   a   c   g   t 
## 173 181 247 188
\end{verbatim}

For GC contents

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{seqinr}\OperatorTok{::}\KeywordTok{GC}\NormalTok{(allocated_Seq)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0.5424588
\end{verbatim}

The length of the sequence is 789 and the GC proportion is 54.24588\%

\#\#\#Question 3 You will be provided with R functions to create BLAST
databases and perform blast searches. Use blast to identify what
\emph{E. coli} gene your sequence matches best. Show a table of the top
3 hits including percent identity, E-value and bit scores. 4 \#\#Answer:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{source}\NormalTok{(}\StringTok{"https://raw.githubusercontent.com/markziemann/SLE712_files/master/bioinfo_asst3_part2_files/mutblast_functions.R"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Blast search

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{res <-}\StringTok{ }\KeywordTok{myblastn_tab}\NormalTok{(allocated_Seq, }\DataTypeTok{db=}\StringTok{"Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.cds.all.fa"}\NormalTok{)}
\NormalTok{res}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   qseqid   sseqid pident length mismatch gapopen qstart qend sstart send evalue
## 1     53 AAC76075    100    789        0       0      1  789      1  789      0
##   bitscore
## 1     1517
\end{verbatim}

We can see that the allocated sequence parfectly matches the Source
\emph{E.coli} sequence. The sequence is ``AAC76075''.

\#The function to calculate the top 3 hit

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{hits <-}\StringTok{ }\KeywordTok{as.character}\NormalTok{(res}\OperatorTok{$}\NormalTok{sseqid[}\DecValTok{1}\OperatorTok{:}\DecValTok{3}\NormalTok{])}
\NormalTok{hits}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "AAC76075" NA         NA
\end{verbatim}

The percent Identity is 100, E- value 0 and Bitscore - 1517

\#\#Question 4 You will be provided with a function that enables you to
make a set number of point mutations to your sequence of interest. Run
the function and write an R code to check the number of mismatches
between the original and mutated sequence.

\#\#Answer: create a copy of my allocated sequnce that has certain
number of mismathces in it

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{seqinr}\OperatorTok{::}\KeywordTok{write.fasta}\NormalTok{(allocated_Seq,}\DataTypeTok{names=}\StringTok{"allocated_Seq"}\NormalTok{,}\DataTypeTok{file.out =} \StringTok{"allocated_Seq.fa"}\NormalTok{)}
\KeywordTok{makeblastdb}\NormalTok{(}\StringTok{"allocated_Seq.fa"}\NormalTok{,}\DataTypeTok{dbtype=}\StringTok{"nucl"}\NormalTok{, }\StringTok{"-parse_seqids"}\NormalTok{)}
\NormalTok{res <-}\StringTok{ }\KeywordTok{myblastn_tab}\NormalTok{(allocated_Seq, }\DataTypeTok{db=}\StringTok{"allocated_Seq.fa"}\NormalTok{)}
\NormalTok{res}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   qseqid        sseqid pident length mismatch gapopen qstart qend sstart send
## 1     53 allocated_Seq    100    789        0       0      1  789      1  789
##   evalue bitscore
## 1      0     1517
\end{verbatim}

changing the number to check the mismatches

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{my_allocated_mutator<-}\KeywordTok{mutator}\NormalTok{(allocated_Seq,}\DecValTok{20}\NormalTok{)}
\NormalTok{res <-}\StringTok{ }\KeywordTok{myblastn_tab}\NormalTok{(my_allocated_mutator, }\DataTypeTok{db=}\StringTok{"allocated_Seq.fa"}\NormalTok{)}
\NormalTok{res}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   qseqid        sseqid pident length mismatch gapopen qstart qend sstart send
## 1     53 allocated_Seq 97.972    789       16       0      1  789      1  789
##   evalue bitscore
## 1      0     1425
\end{verbatim}

\#\#Question 5 Using the provided functions for mutating and BLASTing a
sequence, determine the number and proportion of sites that need to be
altered to prevent the BLAST search from matching the gene of origin.
Because the mutation is random, you may need to run this test multiple
times to get a reliable answer

\#\#Answer : At 1st we need to create a function that will do mutation
and blast and give the result as 0 or 1 depending on the blast was
successful or not

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{myfunc <-}\StringTok{ }\ControlFlowTok{function}\NormalTok{(myseq,nmut) \{ }
\NormalTok{  mutseq <-}\StringTok{ }\KeywordTok{mutator}\NormalTok{( }\DataTypeTok{myseq=}\NormalTok{ myseq , }\DataTypeTok{nmut =}\NormalTok{ nmut) }\CommentTok{#the sequence for mutation, nmut=nmut as it will recognize the number after the sequence as nmut }
\NormalTok{  res <-}\StringTok{ }\KeywordTok{myblastn_tab}\NormalTok{(}\DataTypeTok{myseq=}\NormalTok{ mutseq ,}\DataTypeTok{db=} \StringTok{"Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.cds.all.fa"}\NormalTok{) }\CommentTok{#for blast}
  \ControlFlowTok{if}\NormalTok{ (}\KeywordTok{is.null}\NormalTok{(res)) \{myres=}\StringTok{ }\DecValTok{0}\NormalTok{\} }\ControlFlowTok{else}\NormalTok{ \{myres =}\StringTok{ }\DecValTok{1}\NormalTok{\} }
  \KeywordTok{return}\NormalTok{(myres) }
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

applying the created function

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{myfunc}\NormalTok{(}\DataTypeTok{myseq=}\NormalTok{allocated_Seq,}\DecValTok{789}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0
\end{verbatim}

Then applying the replicate fucntion that will run the same function
several times and that would give a vector of number of values

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{replicate}\NormalTok{(}\DataTypeTok{n =} \DecValTok{100}\NormalTok{, }\KeywordTok{myfunc}\NormalTok{(}\DataTypeTok{myseq=}\NormalTok{ allocated_Seq,}\DecValTok{789}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   [1] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
##  [38] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
##  [75] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
\end{verbatim}

geting the summerised result

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{mean}\NormalTok{(}\KeywordTok{replicate}\NormalTok{(}\DecValTok{100}\NormalTok{,}\KeywordTok{myfunc}\NormalTok{(}\DataTypeTok{myseq =}\NormalTok{ allocated_Seq,}\DecValTok{789}\NormalTok{ )))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0
\end{verbatim}

we need to run this for all the values for nmut. The numbers are saved
as n below which is used to run the replicate command

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{n <-}\KeywordTok{c}\NormalTok{ (}\DecValTok{0}\NormalTok{,}\DecValTok{50}\NormalTok{,}\DecValTok{100}\NormalTok{,}\DecValTok{150}\NormalTok{,}\DecValTok{200}\NormalTok{,}\DecValTok{250}\NormalTok{,}\DecValTok{300}\NormalTok{)}
\NormalTok{myfunction_rep <-}\StringTok{ }\ControlFlowTok{function}\NormalTok{(nmut) \{}
 \KeywordTok{mean}\NormalTok{(}\KeywordTok{replicate}\NormalTok{(}\DecValTok{100}\NormalTok{, }\KeywordTok{myfunc}\NormalTok{(}\DataTypeTok{myseq=}\NormalTok{ allocated_Seq,nmut)))}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

The final answer with sapply comand

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{finalress <-}\StringTok{ }\KeywordTok{sapply}\NormalTok{( n ,myfunction_rep)}
\NormalTok{finalress}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 1.00 1.00 1.00 0.86 0.26 0.03 0.03
\end{verbatim}

\#\#Provide a chart or table that shows how the increasing proportion of
mutated bases reduces the ability for BLAST to match the gene of origin.
Summarise the results in 1 to 2 sentences.

\#\#Answer : The results from previously found results, a chart is
created to show

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Proportions <-}\KeywordTok{c}\NormalTok{ (}\FloatTok{1.00}\NormalTok{, }\FloatTok{1.00}\NormalTok{, }\FloatTok{1.00}\NormalTok{, }\FloatTok{0.79}\NormalTok{, }\FloatTok{0.32}\NormalTok{, }\FloatTok{0.16}\NormalTok{, }\FloatTok{0.00}\NormalTok{)}
\NormalTok{nmut_val <-}\StringTok{ }\KeywordTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{50}\NormalTok{,}\DecValTok{100}\NormalTok{,}\DecValTok{150}\NormalTok{,}\DecValTok{200}\NormalTok{,}\DecValTok{250}\NormalTok{,}\DecValTok{300}\NormalTok{)}
\KeywordTok{plot}\NormalTok{(Proportions,nmut_val,}\DataTypeTok{main=}\StringTok{"How increasing number of bases affects BlAST performnace  "}\NormalTok{,}\DataTypeTok{xlab =} \StringTok{"Prportion of successful BLASTs "}\NormalTok{,}\DataTypeTok{ylab =} \StringTok{"numbers of sites randomised"}\NormalTok{,}\StringTok{"b"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{rmd_files/figure-latex/unnamed-chunk-35-1.pdf} The
proportion of Sucessful BLASTs decreasing as the number of randomised
size increased. It tells the limit of BLAST is when nmut is around 300.

\end{document}
